ui/app/AppLayouts/Chat/ChatColumn/Message.qml

```diff
@@ -186,15 +187,53 @@ Item {
         textFormat: Text.RichText
     }
 
+    StyledText {
+        id: dateGroupLbl
+        font.pixelSize: 13
+        color: Style.current.darkGrey
+        horizontalAlignment: Text.AlignHCenter
+        anchors.horizontalCenter: parent.horizontalCenter
+        text: {
+            if (prevMessageIndex == -1) return ""; // identifier
+
+            let now = new Date()
+            let yesterday = new Date()
+            yesterday.setDate(now.getDate()-1)
+
+            let prevMsgTimestamp = chatsModel.messageList.getMessageData(prevMessageIndex, "timestamp")
+            var currentMsgDate = new Date(parseInt(timestamp, 10));
+            var prevMsgDate = prevMsgTimestamp === "" ? new Date(0) : new Date(parseInt(prevMsgTimestamp, 10));
+            if(currentMsgDate.getDay() != prevMsgDate.getDay()){
+                if (now.toDateString() == currentMsgDate.toDateString()) {
+                    return qsTr("Today")
+                } else if (yesterday.toDateString() == currentMsgDate.toDateString()) {
+                    return qsTrId("yesterday")
+                } else {
+                    const monthNames = [qsTr("January"), qsTr("February"), qsTr("March"), 
+                                        qsTr("April"), qsTr("May"), qsTr("June"),
+                                        qsTr("July"), qsTr("August"), qsTr("September"),
+                                        qsTr("October"), qsTrId("November"), qsTr("December")];
```

Comment:
For the translation script to work correctly, it would be better to set all the `qsTr` on a different line.
Also, I don't know why, but `November` is a `qsTrId` for some reason